### PR TITLE
Add MessageEvent

### PR DESCRIPTION
**Pull request by quanvm**

### CODE REVIEW

**Comment by quanvm** [comment]
Update MqClient::queue / Add new functionality MqClient::publishMessage

`public function queue($body, string $routingKey, array $context = [], $exchange = '')
    {
        $message = new MQMessageEvent($body, $routingKey, $context);

        if ($request = $this->currentRequest()) {
            $message->addContextRequestId($request);
            $message->addContextActorId($request);
        }

        if (!$exchange) {
            $message->setRoutingKey(Queue::WORKER_QUEUE_NAME)
                ->setPayload(['routingKey' => $routingKey, 'body' => $body]);
        }

        $this->publishMessage($message, $exchange);
    }

    public function publishMessage(MessageEvent $message, $exchange = 'events')
    {
        $properties = [
            'content_type'        => 'application/json',
            'application_headers' => new AMQPTable($message->getContext())
        ];
        $this->channel()->basic_publish(
            new AMQPMessage(json_encode($message), $properties),
            $exchange,
            $message->getRoutingKey()
        );

        $this->logger->debug($message->jsonSerialize(), [
            'exchange'   => $exchange,
            'routingKey' => $message->getRoutingKey(),
            'context'    => $message->getContext()
        ]);
    }`

File: message_event/MessageEvent.php [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?php
+
+namespace go1\util\message_event;
+
+use JsonSerializable;
+
+interface MessageEvent extends JsonSerializable
```
Comment (andytruong) [inline message_event/MessageEvent.php]: -> `EventInterface`.

File: message_event/MessageEvent.php [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+{
+    public function getRoutingKey(): string ;
```
Comment (andytruong) [inline message_event/MessageEvent.php]: If we depends on rabbitMQ, its' message already have this method. Good time to discover sf/messenger?

File: message_event/UserMQMessageEvent.php [hunk 3]
```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace go1\util\message_event;
```
Comment (andytruong) [inline message_event/UserMQMessageEvent.php]: create sub-domain: `user`; move this one under that domain.

File: message_event/UserMQMessageEvent.php [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+use Doctrine\DBAL\Connection;
+use go1\util\message_event\pipelines\JWTEmbedded;
+use go1\util\message_event\pipelines\PortalEmbedded;
+use go1\util\user\UserHelper;
+use Symfony\Component\HttpFoundation\Request;
+
+class UserMQMessageEvent extends MQMessageEvent
```
Comment (andytruong) [inline message_event/UserMQMessageEvent.php]: UserMQMessageEvent -> UserEvent.

File: message_event/pipelines/MessageEmbeddedPipeline.php [hunk 5]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+namespace go1\util\message_event\pipelines;
+
+use go1\util\message_event\MQMessageEvent;
+
+interface MessageEmbeddedPipeline
+{
+    public function embed(MQMessageEvent $message): void;
```
Comment (andytruong) [inline message_event/pipelines/MessageEmbeddedPipeline.php]: This method should be part of EventInterface.

**Comment by andytruong** [comment]
Please move this to https://github.com/go1com/util_publishing_event

**Pull request closed without merging**